src/aiida_quantumespresso/workflows/pw/bands.py

```diff
@@ -209,14 +210,46 @@ def run_seekpath(self):
 
         This is only called if the `bands_kpoints` input was not specified.
         """
+
         inputs = {
             'reference_distance': self.inputs.get('bands_kpoints_distance', None),
             'metadata': {
                 'call_link_label': 'seekpath'
             }
         }
         result = seekpath_structure_analysis(self.ctx.current_structure, **inputs)
-        self.ctx.current_structure = result['primitive_structure']
+
+        # If the input structure was a HubbardStructureData
```

Comment:
I haven't check the logic here, but instead of adding this to the `PwBandsWorkChain`, wouldn't it make more sense to move this into the `seekpath_structure_analysis`?

Reply:
I tried the logic, it worked , should i move it to `seekpath_structure_analysis` calcfunction? (I am ok with :D)